src/main/resources/assets/computercraft/lua/rom/programs/shell.lua

```diff
@@ -114,7 +115,22 @@ else
 end
 
 local function run( _sCommand, ... )
-    local sPath = shell.resolveProgram( _sCommand )
+    local checkvar = { ... }
+    table.insert( checkvar, 1, _sCommand )
+    for k, v in ipairs( checkvar ) do
+        if v:find( "$", 1, true ) == 1 then
+            v = v:sub( 2, -1)
+            if type( shellenv[v] ) == "string" or type( shellenv[v] ) == "number" or type( shellenv[v] ) == "boolean" then
+                v = tostring( shellenv[v] )
+            elseif type( shellenv[v] ) == "function" then
+                v = tostring( shellenv[v]() )
```

Comment:
I'm personally not a fan of this - environment variables should just be string -> string mappings, not arbitrary code.